lib/mozart_fetcher/router.ex

```diff
@@ -21,19 +21,36 @@ defmodule MozartFetcher.Router do
 
   post "/collect" do
     {:ok, body, conn} = read_body(conn)
-    body = decode(body)
-    resp = Fetcher.process(body["components"])
+    send_response(config(body), conn)
+  end
+
+  def send_response({:ok, components}, conn) do
+    response = Fetcher.process(components)
+    conn
+    |> put_resp_content_type("application/json")
+    |> send_resp(200, response)
+  end
 
+  def send_response(:error, conn) do
     conn
     |> put_resp_content_type("application/json")
-    |> send_resp(200, resp)
+    |> send_resp(500, "Internal Server Error")
   end
 
   match _ do
     send_resp(conn, 404, "not found")
   end
 
-  def decode(body) do
-    Poison.decode!(body, as: %{"components" => [%Config{}]})
+  def config(body) do
+    IO.inspect(body)
```

Comment:
Just a tiny thing, but I don't think we still need this IO.inspect

Reply:
Thanks